### PR TITLE
Add create message functionallity

### PR DESCRIPTION
**Pull request by juan-apa**
As a user I should be able to send a message in a conversation with any of my compatible targets so I can talk about our common interests

[Trello card](https://trello.com/c/ws2RQiwk)

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -46,15 +46,19 @@ class User < ApplicationRecord
...
   has_many :matches_compatible,
-           foreign_key: :match_compatible,
+           foreign_key: :user_compatible,
            class_name: 'Match',
            dependent: :destroy,
            inverse_of: :target_compatible
 
   include DeviseTokenAuth::Concerns::User
+
+  def matches
```
Comment (mdornel) [inline app/models/user.rb]: Are you using this method? If you think that could be useful in the future keep it, isn't a blocker.

File: spec/requests/messages/create_mesage_spec.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  let!(:target_1) { create :target, user: user_1 }
+  let!(:target_2) do
+    create :target,
+           radius: target_1.radius,
+           latitude: target_1.latitude,
+           longitude: target_1.longitude,
+           topic_id: target_1.topic.id,
+           user: user_2
+  end
+  let!(:conversation_id) { target_2.matches_creators[0].conversation.id }
```
Comment (mdornel) [inline spec/requests/messages/create_mesage_spec.rb]: you can use first here: `target_2.matches_creators.first.conversation.id`

File: spec/requests/messages/create_mesage_spec.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+          subject
+        }.to change(Message, :count).by(1)
+      end
+
+      it 'returns a success code' do
+        subject
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'returns the target' do
```
Comment (mdornel) [inline spec/requests/messages/create_mesage_spec.rb]: Shouldn't be `returns the message` 🤔 ?

File: spec/requests/messages/create_mesage_spec.rb [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    context 'with incorrect params' do
+      context 'with non existant conversation id' do
+        let!(:conversation_id) { 'invalid_conversation_id!' }
+        subject do
+          post api_v1_conversation_messages_path('invalid_conversation_id!'),
+               params: message,
+               headers: headers,
+               as: :json
+        end
```
Comment (santir489) [inline spec/requests/messages/create_mesage_spec.rb]: Why not re-use the previously defined subject ?

**Pull request merged**